keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -2049,6 +2043,56 @@ Unreserve is called in two scenarios:
 
 ### kubelet
 
+#### Communication between kubelet and resource kubelet plugin
+
+Resource kubelet plugins are discovered through the [kubelet plugin registration
+mechanism](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#device-plugin-registration). A
+new "ResourcePlugin" type will be used in the Type field of the
+[PluginInfo](https://pkg.go.dev/k8s.io/kubelet/pkg/apis/pluginregistration/v1#PluginInfo)
+response to distinguish the plugin from device and CSI plugins.
+
+Under the advertised Unix Domain socket the kubelet plugin provides the
+k8s.io/kubelet/pkg/apis/dra gRPC interface. It was inspired by
+[CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md),
+with “volume” replaced by “resource” and volume specific parts removed.
+
+#### Version skew
+
+Previously, kubelet retrieved ResourceClaims and published ResourceSlices on
+behalf of DRA drivers on the node. The information included in those got passed
+between API server, kubelet, and kubelet plugin using the version of the
+resource.k8s.io used by the kubelet. Combining a kubelet using some older API
+version with a plugin using a new version was not possible because conversion
+of the resource.k8s.io types is only supported in the API server and an old
+kubelet wouldn't know about a new version anyway.
+
+Keeping kubelet at some old release while upgrading the control and DRA drivers
+is desirable and officially supported by Kubernetes. To support the same when
+using DRA, the kubelet now leaves ResourceSlice handling (almost) entirely to
+the plugins. The one exception is that it deletes all ResourceSlices on
+startup. This ensures that no pods depending in DRA get scheduled to the node
+until the required DRA drivers have started up again. It also ensures that
+drivers which don't get started up again at all don't leave stale
+ResourceSlices behind. For the same reasons, the ResourceSlices belonging to a
+driver get removed when the driver unregisters. This access is done with
+whatever resource.k8s.io API version is the latest known to the kubelet. To
+support version skew, support for older API versions must be preserved as far
+back as support for older kubelet releases is desired.
```

Comment:
I would not expect this to require preserving alpha versions. Once in beta, as long as kubelets update to use new API versions in the release they are added, the standard 3-release deprecation period before removal of a beta API will cover skewed kubelets.

Reply:
We might want to keep one alpha when graduating to beta because then we can actually test the version skew. We can also do this during the development phase (add beta, test, remove alpha), but perhaps then it's worth keeping?

I don't have a strong preference here.

Reply:
Note that the wording is intentionally so that we can drop alpha support or keep it, depending on what we consider "desired".